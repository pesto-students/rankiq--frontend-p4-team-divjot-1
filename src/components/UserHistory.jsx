import Box from '@mui/material/Box';

function UserHistory() {
  return <Box>User History Route</Box>;
}

export default UserHistory;
