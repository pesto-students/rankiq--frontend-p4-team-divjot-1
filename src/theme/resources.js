import enDefault from './l10n/en.json';
import hiDefault from './l10n/hi.json';

const resources = {
  en: {
    default: enDefault,
  },
  hi: {
    default: hiDefault,
  },
};

export default resources;
