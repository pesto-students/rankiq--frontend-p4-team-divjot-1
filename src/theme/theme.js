import { createTheme } from '@mui/material/styles';

const themeOptions = {
  palette: {
    type: 'light',
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#ffffff',
    },
    text: { navFooter: '#FFFFFF' },
    box: { main: '#FAFAFA' },
  },
};

const theme = createTheme(themeOptions);

export default theme;
