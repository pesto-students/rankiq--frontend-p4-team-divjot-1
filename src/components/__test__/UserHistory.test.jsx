import { beforeEach, describe, expect, it, vi } from 'vitest';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import UserHistory from '../UserHistory';
import theme from '../../theme/theme';

const mockStore = configureStore([thunk]);
const navigate = vi.fn();
vi.mock('react-router-dom', async () => {
  return {
    ...(await vi.importActual('react-router-dom')),
    useNavigate: () => navigate,
  };
});

describe('UserHistory- user history is present', () => {
  let tree;
  let store;
  const storeData = {
    auth: {
      userInfo: { email: 'email', firstName: 'Test Name' },
      accessToken: 'testToken',
    },
    userHistory: {
      loading: false,
      data: [
        { rollNumber: '1234', mark: 50 },
        { rollNumber: '4567', mark: 11 },
      ],
    },
    rankData: {
      loading: false,
      error: null,
      data: {
        overall: { rank: 1, overall: 12 },
        zone: { rank: 1, overall: 12 },
        zoneCategory: { rank: 1, overall: 12 },
        zoneShift: { rank: 1, overall: 12 },
        zoneShiftCategory: { rank: 1, overall: 12 },
        rawMarks: 12.12,
        zoneSelected: 'test zone',
      },
    },
  };
  beforeEach(() => {
    store = mockStore(storeData);
    tree = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <UserHistory />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  });

  it('should render without any errors', () => {
    const { asFragment } = tree;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call navigate on click of result button', () => {
    const resultButton = screen.getByTestId('result1234');
    fireEvent.click(resultButton);
    waitFor(() => {
      expect(navigate).toHaveBeenCalledWith('/result');
    });
  });

  it('should show rank modal when rank button is clicked', () => {
    const rankButton = screen.getByTestId('rank1234');
    fireEvent.click(rankButton);
    waitFor(() => {
      const modalCloseBtn = screen.getByTestId('btnCloseRank');
      expect(modalCloseBtn).toBeInTheDocument();
      fireEvent.click(modalCloseBtn);
      expect(screen.getByTestId('btnCloseRank')).not.toBeInTheDocument();
    });
  });
});

describe('UserHistory- user history is loading', () => {
  let tree;
  let store;
  const storeData = {
    auth: {
      userInfo: { email: 'email', firstName: 'Test Name' },
      accessToken: 'testToken',
    },
    userHistory: {
      loading: true,
      data: [],
    },
  };
  beforeEach(() => {
    store = mockStore(storeData);
    tree = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <UserHistory />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  });

  it('should render without any errors', () => {
    const { asFragment } = tree;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show circular loader when data is being loaded', () => {
    expect(screen.queryByTestId('circularProgressHistory')).toBeInTheDocument();
  });
});

describe('UserHistory- error in loading user history', () => {
  let tree;
  let store;
  const storeData = {
    auth: {
      userInfo: { email: 'email', firstName: 'Test Name' },
      accessToken: 'testToken',
    },
    userHistory: {
      loading: false,
      data: [],
      error: 'Test Error',
    },
  };
  beforeEach(() => {
    store = mockStore(storeData);
    tree = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <UserHistory />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  });

  it('should render without any errors', () => {
    const { asFragment } = tree;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show circular loader when data is being loaded', () => {
    expect(screen.queryByText('Test Error')).toBeInTheDocument();
  });
});

describe('UserHistory- Unauthorized from server', () => {
  let store;
  const storeData = {
    auth: {
      userInfo: { email: 'email', firstName: 'Test Name' },
      accessToken: 'testToken',
    },
    userHistory: {
      loading: false,
      data: [],
      error: 'Unauthorized',
    },
  };
  beforeEach(() => {
    store = mockStore(storeData);
    render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <UserHistory />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  });

  it('should navigate to login page', () => {
    expect(navigate).toHaveBeenLastCalledWith('/login');
  });
});
