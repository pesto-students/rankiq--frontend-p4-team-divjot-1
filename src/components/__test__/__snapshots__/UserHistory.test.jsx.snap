// Vitest Snapshot v1

exports[`UserHistory- error in loading user history > should render without any errors 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Test Error
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserHistory- error loading user history > should render without any errors 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-66s61j"
    data-testid="circularProgressHistory"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-osm0jy-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`UserHistory- user history is loading > should render without any errors 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-66s61j"
    data-testid="circularProgressHistory"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-osm0jy-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`UserHistory- user history is loading > should render without any errors 2`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-66s61j"
    data-testid="circularProgressHistory"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-osm0jy-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`UserHistory- user history is present > should render without any errors 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i3kqnq-MuiPaper-root-MuiCard-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        userHistory.hi
      </h4>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-17subee-MuiTypography-root"
      >
        userHistory.pageInfo
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1e6jbaj-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-fa43rr-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-lg-7 css-1637pp4-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                >
                  1234
                </h6>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-5 css-ht014-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1t3dpe8-MuiTypography-root"
                >
                  userHistory.marks
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-wm6e9j-MuiTypography-root"
                >
                  50
                </h6>
              </div>
            </div>
            <div
              class="MuiBox-root css-9h4zif"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-z634fe-MuiButtonBase-root-MuiButton-root"
                data-testid="rank1234"
                tabindex="0"
                type="button"
              >
                userHistory.checkRank
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-uzt55w-MuiButtonBase-root-MuiButton-root"
                data-testid="result1234"
                tabindex="0"
                type="button"
              >
                userHistory.result
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1e6jbaj-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-fa43rr-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-lg-7 css-1637pp4-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                >
                  4567
                </h6>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-5 css-ht014-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1t3dpe8-MuiTypography-root"
                >
                  userHistory.marks
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-wm6e9j-MuiTypography-root"
                >
                  11
                </h6>
              </div>
            </div>
            <div
              class="MuiBox-root css-9h4zif"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-z634fe-MuiButtonBase-root-MuiButton-root"
                data-testid="rank4567"
                tabindex="0"
                type="button"
              >
                userHistory.checkRank
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-uzt55w-MuiButtonBase-root-MuiButton-root"
                data-testid="result4567"
                tabindex="0"
                type="button"
              >
                userHistory.result
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
